diff --git a/clang-tools-extra/clangd/tool/ClangdMain.cpp b/clang-tools-extra/clangd/tool/ClangdMain.cpp
index 13fe4d3..488069d 100644
--- a/clang-tools-extra/clangd/tool/ClangdMain.cpp
+++ b/clang-tools-extra/clangd/tool/ClangdMain.cpp
@@ -886,12 +886,6 @@ clangd accepts flags on the commandline, and in the CLANGD_FLAGS environment var
   if (Tracer)
     TracingSession.emplace(*Tracer);
 
-  // If a user ran `clangd` in a terminal without redirecting anything,
-  // it's somewhat likely they're confused about how to use clangd.
-  // Show them the help overview, which explains.
-  if (llvm::outs().is_displayed() && llvm::errs().is_displayed() &&
-      !CheckFile.getNumOccurrences())
-    llvm::errs() << Overview << "\n";
   // Use buffered stream to stderr (we still flush each log message). Unbuffered
   // stream can cause significant (non-deterministic) latency for the logger.
   llvm::errs().SetBuffered();
