--- a/clang-tools-extra/clangd/Compiler.cpp
+++ b/clang-tools-extra/clangd/Compiler.cpp
@@ -62,7 +62,6 @@
   // Disable any pch generation/usage operations. Since serialized preamble
   // format is unstable, using an incompatible one might result in unexpected
   // behaviours, including crashes.
-  CI.getPreprocessorOpts().ImplicitPCHInclude.clear();
   CI.getPreprocessorOpts().PrecompiledPreambleBytes = {0, false};
   CI.getPreprocessorOpts().PCHThroughHeader.clear();
   CI.getPreprocessorOpts().PCHWithHdrStop = false;
