diff --git a/lld/tools/lld/CMakeLists.txt b/lld/tools/lld/CMakeLists.txt
index 1d1a7ff4..fa1c7061 100644
--- a/lld/tools/lld/CMakeLists.txt
+++ b/lld/tools/lld/CMakeLists.txt
@@ -27,10 +27,6 @@ endfunction()
 lld_target_link_libraries(lld
   PRIVATE
   lldCommon
-  lldCOFF
-  lldELF
-  lldMachO
-  lldMinGW
   lldWasm
   )
 
diff --git a/lld/tools/lld/lld.cpp b/lld/tools/lld/lld.cpp
index d6800fa1..e912518f 100644
--- a/lld/tools/lld/lld.cpp
+++ b/lld/tools/lld/lld.cpp
@@ -66,12 +66,10 @@ static unsigned inTestVerbosity() {
   return v;
 }
 
-LLD_HAS_DRIVER(coff)
-LLD_HAS_DRIVER(elf)
-LLD_HAS_DRIVER(mingw)
-LLD_HAS_DRIVER(macho)
 LLD_HAS_DRIVER(wasm)
 
+static const lld::DriverDef wasmOnly[] = {{lld::Wasm, &lld::wasm::link}};
+
 int lld_main(int argc, char **argv, const llvm::ToolContext &) {
   sys::Process::UseANSIEscapeCodes(true);
 
@@ -87,7 +85,7 @@ int lld_main(int argc, char **argv, const llvm::ToolContext &) {
   // exception handling and no memory cleanup on exit.
   if (!inTestVerbosity()) {
     int r =
-        lld::unsafeLldMain(args, llvm::outs(), llvm::errs(), LLD_ALL_DRIVERS,
+        lld::unsafeLldMain(args, llvm::outs(), llvm::errs(), wasmOnly,
                            /*exitEarly=*/true);
     return r;
   }
@@ -100,7 +98,7 @@ int lld_main(int argc, char **argv, const llvm::ToolContext &) {
     inTestOutputDisabled = (i != 1);
 
     // Execute one iteration.
-    auto r = lldMain(args, llvm::outs(), llvm::errs(), LLD_ALL_DRIVERS);
+    auto r = lldMain(args, llvm::outs(), llvm::errs(), wasmOnly);
     if (!r.canRunAgain)
       exitLld(r.retCode); // Exit now, can't re-execute again.
 
